Load templates from template engines’ DIRS setting

thibaudcolas opened the pull request:
## Description

This makes the pattern library search for templates both in [`DIRS`](https://docs.djangoproject.com/en/3.2/ref/settings/#dirs) directories for template engines, and in the `templates` subdirectory inside each installed application if using [`APP_DIRS`](https://docs.djangoproject.com/en/3.2/ref/settings/#app-dirs) as it was before.

This is necessary so the package is usable for projects that don’t use APP_DIRS.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added an appropriate CHANGELOG entry

bcdickinson commented on tests/tests/test_utils.py:
```diff
@@ -33,3 +36,40 @@ def test_parent_template_from_variable(self):
+        },
+    ])
+    def test_get_template_dirs_app_dirs(self):
+        template_dirs = ['/'.join(d.replace(os.path.dirname(settings.BASE_DIR), 'dpl').split('/')[-4:-1]) for d in get_template_dirs()]
```
This is a bit opaque. Could we add a comment describing what this line is doing?

thibaudcolas replied:
Indeed 🤦  This does two things:

- Absolute path -> relative path
- Rewrite the current project’s paths so they are normalised regardless of the root folder’s name (dependent on how the git repository is cloned)

bcdickinson commented on tests/tests/test_utils.py:
```diff
@@ -33,3 +36,40 @@ def test_parent_template_from_variable(self):
+        },
+    ])
+    def test_get_template_dirs_list_dirs(self):
+        template_dirs = ['/'.join(d.replace(os.path.dirname(settings.BASE_DIR), 'dpl').split('/')[-4:-1]) for d in get_template_dirs()]
```
Same comment as before, a comment here would be quite handy :smile:

bcdickinson reviewed (commented):
Changes LGTM, thanks Thibaud. I had some very minor feedback about commenting the lines that manipulate the directory lists in the tests.

thibaudcolas commented:
Thank you @bcdickinson! I’ve refactored the tests, and also fixed a compatibility issue with Django 3.2 (`dir` entries are now pathlib.Path instances). I made no changes to the actual implementation so will now merge without further review.
